lib/rspec/support/differ.rb

```diff
@@ -72,6 +76,46 @@ def initialize(opts={})
       end
 
     private
+      def run(actual, expected)
+        diff = ""
+
+        no_procs_no_numbers = lambda {|var1, var2| no_procs?(var1, var2) && no_numbers?(var1, var2)}
```

Comment:
Is there a reason not to make this a method?

Reply:
I don't know, I don't have a strong opinion either here. Originally [I did it this way](https://github.com/rspec/rspec-support/pull/596/commits/d50d7429fe0ac20744477deba49a13d8989d6a3b#diff-df5537736876b545aac78c1d539a1836932f9a2d6c2170e9457490988b47a3f4L22):

```
            if no_procs?(actual, expected) && no_numbers?(actual, expected)
               diff = diff_as_object_with_anything(actual, expected)
             end
          elsif no_procs?(actual, expected) && no_numbers?(actual, expected)
             diff = diff_as_object(actual, expected)
```

But Rubocop complained about offending `Metrics::PerceivedComplexity` rule, in order to lower the perceived complexity score I made the lambda function to reduce the number of `&&` operators by 1, and that made the offense disappear.

I didn't made a method because I didn't know if it was overkill or not to create a method that only me will use it in only two places. But I don't have a strong opinion, I can write the method if you think it is better :)

Reply:
I mean is it possible to extract ‘no_procs_no_numbers?’ to a method? That would even further reduce the complexity, right? And (I may be mistaken) reduce memory allocations on subsequent calls.